src/EditorFeatures/Core/Shared/Tagging/TagSources/ProducerPopulatedTagSource.cs

```diff
@@ -666,7 +677,7 @@ public override ITagSpanIntervalTree<TTag> GetTagIntervalTreeForBuffer(ITextBuff
             TagSpanIntervalTree<TTag> tags;
             if (!map.TryGetValue(buffer, out tags))
```

Comment:
it is not related to paul's change, but not sure why we care about _previewCachedTags or _cachedTags stuff. it should always return up to date result unless we already have one. not sure whether current architecture provide a way to check whether cached tags are up to date.